experimental/FTheoryTools/src/auxiliary.jl

```diff
@@ -170,51 +170,115 @@ function _kodaira_type(id::MPolyIdeal{T}, f::T, g::T, d::T, ords::Tuple{Int64, I
   elseif d_ord >= 12 && f_ord >= 4 && g_ord >= 6
     kod_type = "Non-minimal"
   else
-    # Create new ring with auxiliary variable to construct the monodromy polynomial
-    R = parent(f)
-    S, (_psi, ) = polynomial_ring(QQ, ["_psi"; [string(v) for v in gens(R)]], cached = false)
-    ring_map = hom(R, S, gens(S)[2:end])
-    poly_f = ring_map(f)
-    poly_g = ring_map(g)
-    poly_d = ring_map(d)
-    locus = ring_map(gens(id)[1])
-    
-    # Compute monodromy polynomial and check factorization for remaining cases
-    if f_ord == 0 && g_ord == 0
-      g_quotient = divrem(9 * poly_g, locus)[2]
-      f_quotient = divrem(2 * poly_f, locus)[2]
-      quotient_val = div(g_quotient, f_quotient)
-
-      monodromy_poly = _psi^2 + quotient_val
-      kod_type = _string_from_factor_count(monodromy_poly, ["Non-split I_$d_ord", "Split I_$d_ord"])
-    elseif d_ord == 4 && g_ord == 2 && f_ord >= 2
-      g_quotient = divrem(div(poly_g, locus^2), locus)[2]
-
-      monodromy_poly = _psi^2 - g_quotient
-      kod_type = _string_from_factor_count(monodromy_poly, ["Non-split IV", "Split IV"])
-    elseif d_ord == 6 && f_ord >= 2 && g_ord >= 3
-      f_quotient = divrem(div(poly_f, locus^2), locus)[2]
-      g_quotient = divrem(div(poly_g, locus^3), locus)[2]
-      
-      monodromy_poly = _psi^3 + _psi * f_quotient + g_quotient
-      kod_type = _string_from_factor_count(monodromy_poly, ["Non-split I^*_0", "Semi-split I^*_0", "Split I^*_0"])
-    elseif f_ord == 2 && g_ord == 3 && d_ord >= 7
-      d_quotient = div(poly_d, locus^d_ord)
-      f_quotient = div(2 * poly_f, locus^2)
-      g_quotient = div(9 * poly_g, locus^3)
-      num_quotient = divrem(d_quotient * f_quotient^(2 + d_ord % 2), locus)[2]
-      den_quotient = divrem(4 * g_quotient^(2 + d_ord % 2), locus)[2]
-      quotient_val = div(num_quotient, den_quotient)
-
-      monodromy_poly = _psi^2 + quotient_val
-      kod_type = _string_from_factor_count(monodromy_poly, ["Non-split I^*_$(d_ord - 6)", "Split I^*_$(d_ord - 6)"])
-    elseif d_ord == 8 && g_ord == 4 && f_ord >= 3
-      g_quotient = divrem(div(poly_g, locus^4), locus)[2]
-
-      monodromy_poly = _psi^2 - g_quotient
-      kod_type = _string_from_factor_count(monodromy_poly, ["Non-split IV^*", "Split IV^*"])
+    if concrete_base
+      # Over concrete bases, we reduce to only two variables
+      # so that the computation is faster
+      num_gens = length(gens(parent(f)))
+      first_coord_inds = collect(1:num_gens - 2)
+      last_coord_inds = collect(3:num_gens)
+      rand_ints_ab2 = rand(-100:100, num_gens - 2)
```

Comment:
Are we guaranteed that the method always works? Or could these random numbers sometimes fail us? If so, we would have random failures, which is likely not good. 

(There were discussions elsewhere regarding fixed seeds etc., so as to make tests/failures reproducible. We should investigate if random numbers are truly needed.)

Reply:
Discussed this in triage.

If you really need randomness here, I would strongly suggest to add a comment that explains the background.

Also: are the given ranges really always sufficient? If so this should be justified. If not then one should at least describe what might go wrong if the random value is "insufficient" (whatever that means). You could also add checks or loop to try multiple random values until some criterion proves it is correct; or until a consistent result value has been produced several times for different random choices...

More was said and discussed with @HereAround but that's the gist.

Reply:
@apturner @emikelsons and I just discussed more on this. Here is the summary:

What this function is actually deciding is if certain ideals are radical or not. However, for the geometries encountered in FTheory, these computations are too hard/time consuming. The random numbers serve as shortcut as follows:

1. Keep exactly two variable (in the polynomial generators of the ideals in question), and evaluate all remaining variables at integer positions. Then check whether the result is radical or not.
2. Currently, this step is executed twice - once by keeping the very first two variables as symbols (and evaluating all other variables to random integer values), and once by keeping the very last two variables as symbols.

We just agreed on the following:
* Instead of keep always the first two/last two variables as symbols, do so for random subsets of variables.
* Repeat the evaluation and subsequent radical check five times (instead of currently only two times).
* Add tests, to gauge if this algorithm succeeds.
* The current implementation corrects for a tendency of this algorithm to